src/main/groovy/jaci/gradle/nativedeps/NativeDepsPlugin.groovy

```diff
@@ -225,9 +229,25 @@ class NativeDepsPlugin implements Plugin<Project> {
             )
         }
 
+        private static List<Flavor> getFlavors(BaseLibSpec lib, final FlavorContainer flavors) {
+            if (lib.flavor == null && (lib.flavors == null || lib.flavors.empty))
+                return [null] as List;
+            return (lib.flavors ?: [lib.flavor] as List<String>).collect {
+                flavors.getByName(it) as Flavor
+            }
+        }
+
+        private static List<BuildType> getBuildTypes(BaseLibSpec lib, final BuildTypeContainer buildTypes) {
+            if (lib.buildType == null && (lib.buildTypes == null || lib.buildTypes.empty))
+                return [null] as List;
+            return (lib.buildTypes ?: [lib.buildType] as List<String>).collect {
+                buildTypes.getByName(it) as BuildType
+            }
+        }
+
         private static List<NativePlatform> getPlatforms(BaseLibSpec lib, final PlatformContainer platforms) {
             if (lib.targetPlatform == null && (lib.targetPlatforms == null || lib.targetPlatforms.empty))
-                return [] as List;
+                return [null] as List;

```

Comment:
I'm pretty sure targetPlatform isn't null-safe at other points in the code, particularly in `appliesTo`